crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -51,6 +56,209 @@ use sui_types::sui_serde::{Base64, Encoding};
 mod rpc_types_tests;
 
 pub type GatewayTxSeqNumber = u64;
+pub type SuiMoveTypeParameterIndex = u16;
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveAbilitySet(u8);
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum SuiMoveVisibility {
+    Private = 0x0,
+    Public = 0x1,
+    Friend = 0x3,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct SuiMoveStructTypeParameter {
+    pub constraints: SuiMoveAbilitySet,
+    pub is_phantom: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedField {
+    pub name: String,
+    pub type_: MoveNormalizedType,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub struct MoveNormalizedStruct {
+    pub abilities: SuiMoveAbilitySet,
+    pub type_parameters: Vec<SuiMoveStructTypeParameter>,
+    pub fields: Vec<MoveNormalizedField>,
+}
+
+#[derive(Serialize, Deserialize, Debug, JsonSchema)]
+pub enum MoveNormalizedType {
+    Bool,
+    U8,
+    U64,
+    U128,
+    Address,
+    Signer,
+    Struct {
```

Comment:
I don't know if you can solve this here easily, but the way this gets exposed over the response is kind of weird. It looks like this becomes an object with a `Struct` property on it.

Reply:
this Struct is not the SuiMoveNormalizedStruct, I am not sure why is this on the Move part, but the original definition is here
https://github.com/move-language/move/blob/8958b40e97a4f94a588e5365fdb90ec18f00d888/language/move-binary-format/src/normalized.rs#L32-L45